HaskellEmacs.hs

```diff
@@ -8,17 +9,20 @@ import           Control.Arrow
 import           Control.Concurrent
 import           Control.Monad                    (forever)
 import           Control.Parallel.Strategies
-import           Data.AttoLisp
-import qualified Data.Attoparsec.ByteString.Char8 as AC
+import           Data.AttoLisp                    
+import           Data.Maybe                       (fromJust,fromMaybe, catMaybes)
+import           Data.Monoid                      ((<>))
+import           Data.Text                        (Text, unpack, pack)
+import           Language.Haskell.Exts.Parser     (parseModule, ParseResult(..))
+import           Language.Haskell.Exts.Syntax     (Module(..), ExportSpec(..), QName(..), Name, Decl(FunBind, PatBind), Match(Match), Pat(..))
+import           System.IO                        (hFlush, stdout)
+import qualified Data.Attoparsec.ByteString.Char8 as AC 
 import qualified Data.Attoparsec.ByteString.Lazy  as A
 import qualified Data.ByteString.Lazy.Char8       as B hiding (length)
 import qualified Data.ByteString.Lazy.UTF8        as B (length)
 import qualified Data.Map                         as M
-import           Data.Maybe                       (fromJust,fromMaybe)
-import           Data.Monoid                      ((<>))
-import           Data.Text                        (Text)
 import qualified Data.Text                        as T
-import           System.IO                        (hFlush, stdout)
+import qualified Language.Haskell.Exts.Syntax     as S (Name(Ident, Symbol)) 
 
```

Comment:
No problem (they were already sorted (according to stylish-haskell, alphabetically and ignoring `qualified'))